hazelcast/src/main/java/com/hazelcast/config/AbstractWanPublisherConfig.java

```diff
@@ -0,0 +1,187 @@
+     * {@link WanReplicationConfig}.
+     *
+     * @return the WAN publisher ID or {@code null} if no publisher ID is set
+     */
+    public String getPublisherId() {
+        return publisherId;
+    }
+
+    /**
+     * Sets the publisher ID used for identifying the publisher in a
+     * {@link WanReplicationConfig}.
+     *
+     * @param publisherId the WAN publisher ID
+     * @return this config
+     */
+    public AbstractWanPublisherConfig setPublisherId(String publisherId) {
+        this.publisherId = publisherId;
+        return this;
+    }
+
+    /**
+     * Returns the WAN publisher properties.
+     */
+    public @Nonnull
+    Map<String, Comparable> getProperties() {
+        return properties;
+    }
+
+    /**
+     * Sets the WAN publisher properties.
+     *
+     * @param properties WAN publisher properties
+     * @return this config
+     */
+    public AbstractWanPublisherConfig setProperties(@Nonnull Map<String, Comparable> properties) {
+        this.properties = checkNotNull(properties, "Properties must not be null");
+        return this;
+    }
+
+    /**
+     * Returns the name of the class implementing the WanReplicationEndpoint.
+     * NOTE: OS and EE have different interfaces that this class should implement.
+     * For OS see {@link com.hazelcast.wan.WanReplicationEndpoint}.
+     */
+    public String getClassName() {
+        return className;
+    }
+
+    /**
+     * Sets the name of the class implementing the WanReplicationEndpoint.
+     * NOTE: OS and EE have different interfaces that this class should implement.
+     * For OS see {@link com.hazelcast.wan.WanReplicationEndpoint}.
+     * To configure the built in WanBatchReplication, please use
+     * {@link WanBatchReplicationPublisherConfig} config class.
+     *
+     * @param className the name of the class implementation for the WAN replication
+     * @return this config
+     */
+    public AbstractWanPublisherConfig setClassName(String className) {
+        this.className = className;
+        return this;
+    }
+
+    /**
+     * Returns the implementation of the WanReplicationEndpoint.
+     * NOTE: OS and EE have different interfaces that this object should implement.
+     * For OS see {@link com.hazelcast.wan.WanReplicationEndpoint}.
+     */
+    public Object getImplementation() {
+        return implementation;
+    }
+
+    /**
+     * Sets the implementation of the WanReplicationEndpoint.
+     * NOTE: OS and EE have different interfaces that this object should implement.
+     * For OS see {@link com.hazelcast.wan.WanReplicationEndpoint}.
+     *
+     * @param implementation the implementation for the WAN replication
+     * @return this config
+     */
+    public AbstractWanPublisherConfig setImplementation(Object implementation) {
+        this.implementation = implementation;
+        return this;
+    }
+
+    @Override
+    public int getFactoryId() {
```

Comment:
shouldn't this class force implementers to implement also `getFactoryId` besides `getClassId`?

Reply:
Good point.

Reply:
Addressed.